test/TestApps/ToastNotificationsTestApp/main.cpp

```diff
@@ -26,16 +26,134 @@ bool UnregisterBackgroundActivationTest()
     return true;
 }
 
-winrt::ToastNotification GetToastNotification()
+winrt::ToastNotification CreateToastNotification(winrt::hstring message)
 {
-    winrt::hstring xmlPayload{ L"<toast>intrepidToast</toast>" };
+    winrt::hstring xmlPayload{ L"<toast>" + message + L"</toast>" };
 
     winrt::XmlDocument xmlDocument{};
     xmlDocument.LoadXml(xmlPayload);
 
     return winrt::ToastNotification(xmlDocument);
 }
 
+winrt::ToastNotification CreateToastNotification()
+{
+    return CreateToastNotification(L"intrepidToast");
+}
+
+void EnsureNoActiveToasts()
+{
+    auto removeAllAsync = winrt::ToastNotificationManager::Default().RemoveAllAsync();
+    if (removeAllAsync.wait_for(std::chrono::seconds(300)) != winrt::Windows::Foundation::AsyncStatus::Completed)
+    {
+        removeAllAsync.Cancel();
+        THROW_HR_MSG(E_FAIL, "Failed to remove all active toasts");
+    }
+}
+
+bool VerifyToastIsActive(UINT32 expectedToastId)
+{
+    auto retrieveNotificationsAsync{ winrt::ToastNotificationManager::Default().GetAllAsync() };
+    if (retrieveNotificationsAsync.wait_for(std::chrono::seconds(300)) != winrt::Windows::Foundation::AsyncStatus::Completed)
+    {
+        retrieveNotificationsAsync.Cancel();
+        return false;
+    }
+
+    auto notifications = retrieveNotificationsAsync.get();
+
+    bool found{ false };
+    for (auto notification : notifications)
+    {
+        if (notification.ToastId() == expectedToastId)
+        {
+            found = true;
+            break;
+        }
+    }
+
+    return found;
+}
+
+enum class ExpectedTransientProperties { EQUAL, DEFAULT, UNKNOWN };
```

Comment:
Can you add a comment describing when to use each of these values? 😊

Reply:
Done